src/test/java/com/mael/ttt/BoardTest.java

```diff
@@ -39,7 +47,7 @@ public void resetsBoard() {
     @Test
     public void getsTheEmptyCellIndexes() {
         board.setCell(1, PLAYER);
-        List<Integer> expected = asList(2, 3, 4, 5, 6, 7, 8, 9);
+        List<Integer> expected = getCellIndexesStartingOn(2);
```

Comment:
Who says you need to to that? ;)
Of course, that's not what anyone wants. Toggling the `size` variable in `setup` every time you want to add something.
That is why I was asking which methods do actually need to change to work with a new board size.

What is so bad about the approach with just instantiating a 4x4 board in a test and have some assertions on it?

> All the test methods should pass no matter the board size

That is correct. Yet think about if all test methods should actually know or care about the board size at all.
The thing is, you created to board from the very beginning in a generic way that would work with any size. Now that we actually introduce a new size, a whole lot of other unrelated classes (clients of `Board`) need to change. And I'm asking myself: why? Why is that needed when the board was capable representing different sizes already.

Shouldn't it be enough to instantiate a 4x4 with a given layout pass it to a `BoardChecker` and verify that it can identify a winner there, too?

It feels like very defensive programming to have _all_ board related tests automatically execute twice. I'd like to know: what is it you fear to break?

Test code is just as important as production code and all the rules, principles and good practices apply just as like. And this test setup with an inheritance hierarchy feels a bit "too smart" to me.